### ISSUE TITLE
HTTP status 404 not found

### ISSUE THREAD

**Issue by will287** [securestate/king-phisher#289]
## Issue Description
When I perform the URL verify test I get a 404 error message.

## Reproduction Steps
1. Start King Phisher starts up with not visible issues
2. When I am configuring a campaign I attempt to verify URL then I get the 404 error 
## Environment Details

Host OS: Kali Linux king phisher version: 1.7.1 python version: 3.6.5
Linux kali.XXX.com 4.14.0-kali2-amd64 #1 SMP Debian 4.14.17-1 kali1 (2018-02-16) x86_64 GNU/Linux
King Phisher Version: 1.7.1 Python version 3.6.5

## Error Details / Stack Trace
Everything appears to be working correctly. I have ran wireshark on the Kali box I can see clean hits on port 80 coming for the client box. It appears to be something with the URL for the host site located on the kali box. I can ping the box using the FQDN or the ip address. I can change the error message if I change the url location to /var/www/html/index.html   Some time it will read unable to connect. 

I have set the UID to both true or false same results 404 error. 
I have attached the DEBUG output

Any help would be greatly appreciated

[KP_404_error.pdf](https://github.com/securestate/king-phisher/files/1882209/KP_404_error.pdf)
. 
```
If you are opening an issue regarding the client, copy text from the "Error
Details" of the error dialog in the client here. If you are opening an issue
regarding the server copy the relevant lines from the logs here.
```

**Comment by zeroSteiner** [comment]
Well you're getting 404 errors so that's a good sign. In your server config file, what is the value of `web_root`, `require_id`, and `vhost_directories`? The issue probably lies somewhere in the configuration there. Also where *should* the index.html file be located?

**Comment by will287** [comment]
I have DNS entries that read phistrain.domainame.com this equal IP address 172.1.1.190
Configuration file server_config.yml contains the below entry 
vhost_directories: false 
web_root: /var/www/ 
My URL should read : http://phistrain.domainame.com/var/www/index.html

**Comment by zeroSteiner** [comment]
If your URL is `http://phistrain.domainame.com/var/www/index.html` and your `web_root` is `/var/www/` Then you should have a file at `/var/www/var/www/index.html` which I doubt is what you intended. You need to either update your URL or create the second var and www directories. The way you're constructing your URL seems like you're trying to use your root folder as the webroot which would be ill-advised to say the least.

**Comment by will287** [comment]
Are you saying my url should be http://phishtrain.domainame.com/index.html

**Comment by zeroSteiner** [comment]
Well you never answered my question as to where `index.html` is on disk but if the answer is that it's at `/var/www/index.html` then yes that should be your URL.

This is also assuming that you left `vhost_directories` as `false`. You also never answered the question on what the `require_id` setting is so I'm working off of limited information here.

**Comment by zeroSteiner** [comment]
Then yes, your URL should be http://phishtrain.domainame.com/index.html

**Comment by will287** [comment]
Thank you that makes scenes. I am not using the above example as my url  to verify and I am still getting a 404 error although it appears to be a little different the error is as follows : 
jinja2 template /var/www/index.html render failed: UndefinedError 'url' is undifined

**Comment by zeroSteiner** [comment]
As the error indicates, `url` is undefined. Find where you're using it in the jinja template and fix it. The variable reference is [here](https://github.com/securestate/king-phisher/wiki/Templates) in the wiki.

**Comment by will287** [comment]
Sorry I am using the url you described  http://phishtrain.domainame.com/index.html 
The above example as my url to verify and I am still getting a 404 error although it appears to be a little different the error is as follows :
jinja2 template /var/www/index.html render failed: UndefinedError 'url' is undifined

**Comment by will287** [comment]
Checking

**Comment by zeroSteiner** [comment]
Yes, you're getting a 404 error because the template is broken. The King Phisher server can't render it so it responds with the 404 status code. The King Phisher server uses the 404 status code for many things other than just a missing file to prevent information leakage.

**Comment by will287** [comment]
OK thank you. 
First pass not seeing issue with the template I even used a second template. 
Let me use the weekend to do some homework. 
Thanks again

**Comment by will287** [comment]
Thank you for the help. Just needed a clean reboot after you got me on the right track.
